### DISCUSSION TITLE
How can nicegui freely divide areas vertically by raster?

### DISCUSSION THREAD

**Discussion by ponponon** [zauberzeug/nicegui#2230] [Q&A]
### Question

There is the following code for vue and Ant Design Vue

```vue
<template>
    <a-row class="section">
        <a-col :span="24">col</a-col>
    </a-row>
    <a-row class="section-light">
        <a-col :span="12">col-12</a-col>
        <a-col :span="12">col-12</a-col>
    </a-row>
    <a-row class="section-dark">
        <a-col :span="16">col-16</a-col>
        <a-col :span="4">col-4</a-col>
        <a-col :span="4">col-4</a-col>
    </a-row>
    <a-row class="section-gray">
        <a-col :span="1">col-1</a-col>
        <a-col :span="11">col-11</a-col>
        <a-col :span="2">col-2</a-col>
        <a-col :span="10">col-10</a-col>
    </a-row>
</template>

<style>
.section {
    background-color: #f0f0f0;
}

.section-light {
    background-color: #d3d3d3;
}

.section-dark {
    background-color: #a9a9a9;
}

.section-gray {
    background-color: #808080;
}

.section>div,
.section-light>div,
.section-dark>div,
.section-gray>div {
    padding: 8px;
    border: 1px solid #000;
    box-sizing: border-box;
}
</style>

```

<img width="1440" alt="图片" src="https://github.com/zauberzeug/nicegui/assets/38725104/f0c78ade-7a58-4488-bd79-ca12a0a10e5e">

How does nicegui achieve the above?

As far as I know, with nicegui's ui.grid, you can only split a row evenly, for example, if you split a row into 2, you use ui.grid(columns=2), but if I want to split a row into 4, the first one occupies 1/24; the second one occupies 11/24; the third one occupies 2/24; and the fourth one occupies 10/24; How can I achieve this?

**Comment by falkoschindler** [comment]
Here is an example for grid cells that span more than one column:
```py
with ui.grid(columns=24).classes('w-full gap-0'):
    ui.label('col').classes('col-span-full border')
    ui.label('col-12').classes('col-span-12 border')
    ui.label('col-12').classes('col-span-12 border')
    ui.label('col-16').classes('col-[span_16] border')
    ui.label('col-4').classes('col-span-4 border')
    ui.label('col-4').classes('col-span-4 border')
    ui.label('col-1').classes('col-span-1 border')
    ui.label('col-11').classes('col-span-11 border')
    ui.label('col-2').classes('col-span-2 border')
    ui.label('col-10').classes('col-span-10 border')
```

<img width="804" alt="Screenshot 2023-12-23 at 15 11 36" src="https://github.com/zauberzeug/nicegui/assets/5767091/3cd2a2d6-ba22-47ac-ad33-08563b01af74">

Note that there is [no Tailwind class for spanning 16 columns](https://tailwindcss.com/docs/grid-column), but with square brackets we can set [arbitrary values](https://tailwindcss.com/docs/grid-column#arbitrary-values). Alternatively you could use the corresponding CSS definition: `.style('grid-column: span 16 / span 16')`.

**Reply by falkoschindler** [reply]
I added a similar demo to the documentation: https://github.com/zauberzeug/nicegui/commit/6d31dd282ca2b2fb2c209c96f609bfa7adbaed51

**Reply by ponponon** [reply]
Hi, why is it written as col-[span_16] instead of col-span-16? Is it because of the limitations of the Tailwind CSS framework?

**Reply by falkoschindler** [reply]
Yes, as I mentioned, Tailwind has no "col-span-16" class. Usually you can put custom values in square brackets. If you want a width of 42px, you can write "w-[42px]". But for col-span it seems to be different: "col-span-[16]" didn't work, so I read about [arbitrary col-span values](https://tailwindcss.com/docs/grid-column#arbitrary-values) and found the expression "col-[span_16]" with a bit trial and error.
Long story short: It's a Tailwind feature/limitation, so you'll find more detailed explanations in their documentation and forums.
